funnel/models/project_membership.py

```diff
@@ -0,0 +1,187 @@
+# -*- coding: utf-8 -*-
+
+from sqlalchemy.ext.associationproxy import association_proxy
+from sqlalchemy.ext.declarative import declared_attr
+
+from coaster.sqlalchemy import immutable, with_roles
+
+from . import db
+from .membership import ImmutableMembershipMixin
+from .project import Project
+from .user import User
+
+__all__ = ['ProjectCrewMembership']
+
+
+class ProjectCrewMembership(ImmutableMembershipMixin, db.Model):
+    """
+    Users can be crew members of projects, with specified access rights.
+    """
+
+    __tablename__ = 'project_crew_membership'
+
+    # List of is_role columns in this model
+    __data_columns__ = ('is_editor', 'is_concierge', 'is_usher')
+    __parent_column__ = 'project_id'
+
+    __roles__ = {
+        'all': {
+            'read': {
+                'user_details',
+                'is_editor',
+                'is_concierge',
+                'is_usher',
+                'project',
+            },
+            'call': {'url_for'},
+        },
+        'editor': {'read': {'edit_url', 'delete_url'}},
+    }
+
+    project_id = immutable(
+        db.Column(None, db.ForeignKey('project.id', ondelete='CASCADE'), nullable=False)
+    )
+    project = immutable(
+        db.relationship(
+            Project,
+            backref=db.backref(
+                'crew_memberships',
+                lazy='dynamic',
+                cascade='all, delete-orphan',
+                passive_deletes=True,
+            ),
+        )
+    )
+    parent = immutable(db.synonym('project'))
+
+    user_id = immutable(
```

Comment:
We will need the backref when we introduce the tab in user account view showing which projects they are a member of. It could be done without backref as well though.